app/models/host_status/compliance_status.rb

```diff
@@ -0,0 +1,46 @@
+module HostStatus
+  class ComplianceStatus < Status
+    COMPLIANT = 0
+    INCONCLUSIVE = 1
+    INCOMPLIANT = 2
+
+    def self.status_name
+      N_('Compliance')
+    end
+
+    def to_label(options = {})
+      case to_status
+      when COMPLIANT
+        N_('Compliant')
+      when INCONCLUSIVE
+        N_('Inconclusive')
+      when INCOMPLIANT
+        N_('Incompliant')
+      else
+        N_('Unknown Compliance status')
+      end
+    end
+
+    def to_global(options = {})
+      case to_status
+      when COMPLIANT
+        HostStatus::Global::OK
+      when INCONCLUSIVE
+        HostStatus::Global::WARN
+      else
+        HostStatus::Global::ERROR
+      end
+    end
+
+    def relevant?
+      !host.policies.empty?
+    end
+
+    def to_status(options = {})
+      latest_reports = host.policies.map { |p| host.last_report_for_policy p }.flatten.partition { |report| report.failed? }
```

Comment:
`host.policies.map { |p| host.last_report_for_policy p }`

this should be possible to write using SQL, maybe we just need `has_many :arf_reports, :through => :policies` in host? I don't expect too many policies for a single host but still, it's n+1 problem